project_sale_link/models/project_project.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2022 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+from odoo.osv import expression
+from odoo.tools.safe_eval import safe_eval
+
+
+class ProjectProject(models.Model):
+
+    _inherit = "project.project"
+
+    sale_order_count = fields.Integer(
+        compute="_compute_sale_order_count", string="# of Sale Order"
+    )
+
+    customer_invoice_count = fields.Integer(
+        compute="_compute_customer_invoice_count", string="# of Customer Invoice"
+    )
+
+    def _compute_sale_order_count(self):
+        group_data = self.env["sale.order"].read_group(
+            self._get_sale_order_domain(), ["project_id"], ["project_id"]
+        )
+        count_by_id = {
+            data["project_id"][0]: data["project_id_count"] for data in group_data
+        }
+        for rec in self:
+            rec.sale_order_count = count_by_id.get(rec.id, 0)
+
+    def action_open_sale_orders(self):
+        self.ensure_one()
+        action = self.env.ref("sale.action_quotations_with_onboarding").read([])[0]
+        action.update(
+            {
+                "domain": self._get_sale_order_domain(),
+                "context": {"default_project_id": self.id},
+            }
+        )
+        return action
+
+    def _get_sale_order_domain(self):
+        return [("project_id", "in", self.ids)]
+
+    def _compute_customer_invoice_count(self):
+        for project in self:
+            domain = project._get_sale_invoice_line_domain()
+            groups = self.env["account.move.line"].read_group(
+                domain, ["move_id"], ["move_id"],
```

Comment:
Not possible to do this `read_group(...)` before the loop?
You group by `analytic_account_id` then you `sum(...)` results?